### PR TITLE
only prompt user to save if project has changed and won’t autosave

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves

- Resolves https://github.com/LLK/scratch-gui/issues/3744

### Proposed Changes

don't warn user with a confirm about leaving project, if user could choose to save the project, or project will autosave, or project is already saved.

### Reason for Changes

Prompt was sometimes unnecessary and confusing.

### Test Coverage

none.

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Safari
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

### CODE REVIEW

**Comment by paulkaplan** [comment]
@benjiwheeler the code LG, but since the flag you are looking to use is introduced in https://github.com/LLK/scratch-gui/pull/3879, you might just want to wait for that to go in and update this PR, instead of having to make a follow-up PR.

**Comment by benjiwheeler** [comment]
Take a look now!

**Comment by benjiwheeler** [comment]
Hm, projectChanged is making use always warn the user... which makes this little better than before

File: src/components/menu-bar/menu-bar.jsx [hunk 1]
```diff
@@ -726,6 +728,7 @@ MenuBar.propTypes = {
     onShare: PropTypes.func,
     onToggleLoginOpen: PropTypes.func,
     onUpdateProjectTitle: PropTypes.func,
+    projectIsDirty: PropTypes.bool,
```
Comment (paulkaplan) [inline src/components/menu-bar/menu-bar.jsx]: Probably want this to be projectChanged?

**Comment by benjiwheeler** [comment]
Added several tests of when we will and won't see this confirm.

**Pull request merged**